scripts/apiserver/resources.tmpl

```diff
@@ -40,7 +40,9 @@ func init() {
 type {{.res.kind}} struct {
 	apiv1.ResourceMeta
 	{{range $field, $isNotEmpty := .res.fields}}
-		{{$field | strings.Title}} {{if $isNotEmpty}} {{$.res.kind | strings.ReplaceAll "API" "Api" | strings.ReplaceAll "AWS" "Aws"}}{{$field | strings.Title}} {{else}} struct{} {{end}} `json:"{{$field | strings.ToLower}}"`
+		{{ $type := $field | strings.Title }}
+		{{ $varName := $.res.kind | strings.ReplaceAll "API" "Api" | strings.ReplaceAll "AWS" "Aws" }}
+		{{	$type }} {{if $isNotEmpty}} {{$varName}}{{$type}} {{else}} interface{} {{end}} `json:"{{$field | strings.ToLower}}"`
```

Comment:
Why do we need interface{} instead fo struct{} ?

Reply:
Correct me if I am wrong, but an empty struct says we do not want to store any of this data.  So when we unmarshal json into an empty struct we lose that data.  If we unmarshal it into an interface we retain the data and then can later cast it into the appropriate structure.